### ISSUE TITLE
Store returned by GenericAnnotatedTypeFactory.getStoreAfter() sometimes missing expected variables

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#3566]
This issue was observed within the Object Construction Checker.  Example:
```java
class Test {
    class Foo {
    }
    Foo makeFoo(){
        return new Foo();
    }
    void test() {
        Foo foo = makeFoo();
    }
}
```
We are overriding `GenericAnnotatedTypeFactory#postAnalyze` and analyzing the CFG for `test()`.  Given the `AssignmentNode` `n` for `Foo foo = makeFoo()`, we invoke `GenericAnnotatedTypeFactory.getStoreAfter(n)` and expect to find an entry in the `CFStore` for `foo`.  But, `foo` is missing from the store.  Interestingly, if we invoke `getAnnotatedType(n.getTree())` we do get the expected type for `foo`.  We would expect this same type to be in the store immediately after the statement.

To reproduce, check out this branch:

https://github.com/msridhar/object-construction-checker/tree/repro-store-bug

And then run:
```
./gradlew :object-construction-checker:test --tests AlwaysCallTest
```
The test will fail with a RuntimeException that is thrown when the above case is encountered.
